akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerSpec.scala

```diff
@@ -1290,6 +1290,50 @@ class HttpServerSpec extends AkkaSpec(
       netOut.expectComplete()
     })
 
+    "reject requests with an invalid URI schema" in assertAllStagesStopped(new TestSetup {
+      send("""GET htp://www.example.com:80 HTTP/1.1
+             |Host: www.example.com:80
+             |
+             |""")
+
+      requests.request(1)
+
+      expectResponseWithWipedDate(
+        """|HTTP/1.1 400 Bad Request
+           |Server: akka-http/test
+           |Date: XXXX
+           |Connection: close
+           |Content-Type: text/plain; charset=UTF-8
+           |Content-Length: 64
+           |
+           |`uri` must have scheme "http", "https", "ws", "wss" or no scheme""")
+
+      netIn.sendComplete()
+      netOut.expectComplete()
+    })
+
+    "reject HTTP/1.1 requests with invalid Host header schema" in assertAllStagesStopped(new TestSetup {
```

Comment:
The name of the test seems slightly off.

`"reject HTTP/1.1 requests with Host header that doesn't match absolute request target authority"`